### PR TITLE
Add plugin to check apache and proprietary license header

### PR DESCRIPTION
**Pull request by m4dcoder**
Add plugin and related unit tests to check apache and proprietary license header. Enable the license check extension for this repo.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -1,4 +1,5 @@
 [flake8]
 max-line-length = 100
+enable-extensions = L101,L102
```
Comment (Kami) [inline .flake8:3]: It's enabled by default so we don't need this line, but it doesn't hurt either.

**Review by Kami — APPROVED** [review]

File: .flake8 (hunk 1 above)
Comment (m4dcoder) [reply .flake8:3]: Let me test some more.  I added `off_by_default = True` in the class.

Comment (Kami) [reply .flake8:3]: :+1:

Comment (m4dcoder) [reply .flake8:3]: I tested this. Need to explicitly add `enable-extensions = L101,L102` or `enable-extensions = L` in .flake8 config to enable this plugin.

Comment (Kami) [reply .flake8:3]: Yeah, I originally missed ``off_by_default`` line so all is well.

**Pull request merged**